Common/itkComputeImageExtremaFilter.hxx

```diff
@@ -56,9 +56,12 @@ template <typename TInputImage>
 void
 ComputeImageExtremaFilter<TInputImage>::SameGeometry()
 {
-  if (this->GetInput()->GetLargestPossibleRegion().GetSize() ==
-        this->m_ImageSpatialMask->GetImage()->GetLargestPossibleRegion().GetSize() &&
-      this->GetInput()->GetOrigin() == this->m_ImageSpatialMask->GetImage()->GetOrigin())
+  const ImageBase<ImageDimension> & inputImage = *(this->GetInput());
+  const ImageBase<ImageDimension> & maskImage = *(this->m_ImageSpatialMask->GetImage());
+
+  if (inputImage.GetLargestPossibleRegion() == maskImage.GetLargestPossibleRegion() &&
+      inputImage.GetOrigin() == maskImage.GetOrigin() && inputImage.GetSpacing() == maskImage.GetSpacing() &&
+      inputImage.GetDirection() == maskImage.GetDirection())
```

Comment:
Using `==` to compare floating point numbers may be a bit too strict, but it's hard to choose an acceptable error margin for "almost equal". In this case I guess we could compare the coordinates of the corresponding corner points of the image and the mask. Too much work, I guess 😸